.travis.yml

```diff
@@ -1,91 +1,59 @@
-language: python
+language: minimal
+dist: xenial
+services: docker
 
-install:
-    - sudo apt-get -y -q update
-    - sudo apt-get -y -q build-dep herbstluftwm
-    - sudo apt-get -y -q install cmake ninja-build dzen2 xdotool
+
+before_install:
+    # Delete .git/ to maximize ccache effectiveness (no version string mismatches)
+    - rm -rf .git
 
 jobs:
     include:
-        - name: "Ubuntu 16.04, gcc-8, Debug build, run tests"
-          dist: xenial
-          python: "3.6"
-          before_install:
-              - ./ci-check-using-std.sh
-              - export CC=gcc-8 CXX=g++-8
-              - linkerfix="-fuse-ld=gold" # needed to avoid linker error
-              - export LDFLAGS="$linkerfix" LDXXFLAGS="$linkerfix"
-          addons:
-              apt:
-                  sources:
-                      - ubuntu-toolchain-r-test
-                  packages:
-                      - g++-8
-                      - xterm
-                      - tox
+        - name: "Build with most recent GCC & Clang, run checks and tests"
+          env: TRAVIS_CACHE_ID=disco
+          cache:
+              directories:
+                  - $HOME/.ccache-clang-7
+                  - $HOME/.ccache-gcc-8
+                  - $HOME/.tox-cache
           script:
-              - cmake -G'Ninja' -DCMAKE_BUILD_TYPE=Debug . # run tests with sanitizer
-              - ninja -v -k 10
-              - tox -- -n auto -v
+              # Prepare Docker image with bleeding edge Ubuntu 19.04 (disco)
+              - docker pull dnnr/hlwm-ci:disco
+              - docker build -t disco --cache-from dnnr/hlwm-ci:disco - <Dockerfile.ci-disco
 
-        - name: "Ubuntu 16.04, gcc-7"
-          dist: xenial
-          before_install:
-              - export CC=gcc-7 CXX=g++-7
-          addons:
-              apt:
-                  sources:
-                      - ubuntu-toolchain-r-test
-                  packages:
-                      - g++-7
-          services:
-            - xvfb
-          script:
-            - cmake -G'Ninja' -DCMAKE_INSTALL_PREFIX=/usr .
-            - sudo ninja -v -k 10 install
-            - herbstluftwm &
-            - sleep 1
-            - herbstclient version
-            - herbstclient quit
-            - wait
+              # Run builds and tests in Docker container
+              - docker run -u $UID --privileged -t --rm
+                --volume=/etc/passwd:/etc/passwd:ro
+                --volume=$HOME/.ccache-gcc-8:$HOME/.ccache-gcc-8:rw
+                --volume=$HOME/.ccache-clang-7:$HOME/.ccache-clang-7:rw
+                --volume=$PWD:/hlwm:rw
+                --volume=$HOME/.tox-cache:/hlwm/.tox:rw
+                disco
```

Comment:
What is disco?

Reply:
It's the name of the upcoming Ubuntu release: Disco Dingo

Reply:
Oh and in case that wasn't the question: It's also the name of the temporary Docker image that is being used for this `docker run` invocation.